### PR TITLE
Switch from sub ratio to risk buffer for tranche constraints

### PR DESCRIPTION
**Pull request by branan**
Min risk buffer is defined as `<subordinate tranche sum>/<total value>`.

The previous definition was `<subordinate tranche sum>/<tranche value>`.

### CODE REVIEW

**Comment by branan** [comment]
This should rebase cleanly to parachain once TOCC and parachain are merged

File: pallets/tinlake-investor-pool/src/lib.rs [hunk 1]
```diff
@@ -890,28 +890,37 @@ pub mod pallet {
...
+			for (tranche_value, min_risk_buffer) in current_tranche_values
+				.iter()
+				.zip(min_risk_buffers.iter().copied())
+			{
+				buffer_value = buffer_value
+					.checked_sub(tranche_value)
+					.ok_or(Error::<T>::Overflow)?;
+				let risk_buffer = Perquintill::from_rational(buffer_value, total_value);
+				if risk_buffer < min_risk_buffer {
+					Err(Error::<T>::RiskBufferViolated)?
```
Comment (mustermeiszer) [inline pallets/tinlake-investor-pool/src/lib.rs:923]: How do we ensure we are not getting in this state?

**Review by mustermeiszer — APPROVED** [review]
LGTM! 

Just one question regarding how we prevent getting into an `Error::<T>::RiskBufferViolated` state

File: pallets/tinlake-investor-pool/src/lib.rs (hunk 1 above)
Comment (branan) [reply pallets/tinlake-investor-pool/src/lib.rs:923]: This is used as part of validating solutions (and should be renamed or at least documented as such). The solution submitted will need to properly limit the investments and redemptions of the different tranches to keep the constraints satisfied.

**Comment by branan** [comment]
Rebased for conflict with #559. No other changes, so I'll still be merging based on the existing approval once CI is happy

**Pull request merged**